includes/rest-api/class-sensei-rest-api-question-helpers-trait.php

```diff
@@ -185,6 +185,9 @@ private function save_question( $question, $status = 'publish' ) {
 
 		if ( $status ) {
 			$post_args['post_status'] = $status;
+		} elseif ( ! $is_new ) {
+			// If status is not provided, use the current status of the question.
+			$post_args['post_status'] = get_post_status( $question_id );
```

Comment:
Looks like this is a known issue caused by using `wp_insert_post` for [updating posts](https://github.com/Automattic/sensei/blob/5db85c5f041a2ebadc5596b9603ccbc4a66789e4/includes/rest-api/class-sensei-rest-api-question-helpers-trait.php#L122) instead of `wp_update_post` https://developer.wordpress.org/reference/functions/wp_insert_post/#comment-5150.

So I was just wondering, instead of fetching the `status` manually here and assigning it, do you think it'd be a better and more understandable solution to use the `wp_update_post` function when we are actually just updating the post. We'll be using the right function for the right job.

So we just remove the line above and we replace Line 199 with something like `$result = $is_new ? wp_insert_post( $post_args ) : wp_update_post( $post_args );`.

WDYT?

Reply:
Thanks @Imran92!

Yes, it makes sense for the sake of simplicity.

Thinking about it further, I see a little sense of using a ternary. We can simply use `wp_update_post` (`wp_update_post` calls `wp_insert_post` under the hood).

Reply:
Updated here: https://github.com/Automattic/sensei/pull/7603/commits/969082ebfdf9b3790eb66d97ba373a6337e809f3

It isn't that important for the question category, but updated it for consistency: https://github.com/Automattic/sensei/pull/7603/commits/bfd0375a6409b233cf05fc680e9ae17dba49154e